src/schedule/ll_schedule.c

```diff
@@ -265,23 +281,17 @@ static void schedule_ll_domain_clear(struct ll_schedule_data *sch,
 
 	spin_lock(&sch->domain->lock);
 
-	count = atomic_sub(&sch->domain->total_num_tasks, 1);
-	if (count == 1) {
-		domain_clear(sch->domain);
-		sch->domain->last_tick = 0;
-	}
+	atomic_sub(&sch->domain->total_num_tasks, 1);
 
 	count = atomic_sub(&sch->num_tasks, 1);
 	if (count == 1) {
 		sch->domain->registered[cpu_get_id()] = false;
 
-		/* reschedule if we are the last client */
-		if (atomic_read(&sch->domain->num_clients)) {
-			count = atomic_sub(&sch->domain->num_clients, 1);
-			if (count == 1) {
-				domain_clear(sch->domain);
-				schedule_ll_clients_reschedule(sch);
-			}
+		count = atomic_sub(&sch->domain->num_clients, 1);
```

Comment:
Btw, do we have lIRQs OFF here ? I would also check we have this around all the atomic ops as I suspect we dont have full coverage.

Reply:
> Btw, do we have lIRQs OFF here ? I would also check we have this around all the atomic ops as I suspect we dont have full coverage.

We have irq_local_disable in the caller schedule_ll_task_cancel(). I think we are good here, since we have the unique interface for the accessing so the same interrupt is already disabled at the handler entry, the same interrupt have no chance to arrive before we finish this one.